salt/minion.py

```diff
@@ -3199,6 +3195,17 @@ class Matcher(object):
     def __init__(self, opts, functions=None):
         self.opts = opts
         self.functions = functions
+        Matcher._get_matchers_from_loader(opts)
+
+    @classmethod
+    def _get_matchers_from_loader(cls, opts):
+        matcher_functions = salt.loader.matchers(opts)
+        for meth in matcher_functions:
+            function_parts = meth.split('.')
+            function_name = function_parts[0]
+            if not getattr(cls, function_name, False):
+                setattr(cls, function_name, matcher_functions[meth])
```

Comment:
Why do this instead of just calling the matcher_functions[meth] directly?

Reply:
I think only because I felt like it would require less refactoring since the matcher functions were methods on the Matcher object.

Reply:
That is what i guess, this is good enough for a first pass, but in the future we should refactor all of the other places where the matchers are used. :)

Reply:
At least in this case `hasattr` can do it.